### PR TITLE
Ability to Exclude Files/Folders when using Recursive Mode

### PR DESCRIPTION
**Pull request by kaizen365**
Sometimes I want to exclude certain files and/or folders from being compiled when using recursive mode.
This may be due to not wanting the contents of bin/obj folders being compiled, client libraries via libman compiled or even scss files which only contain variables (thus throwing an content error).

### CODE REVIEW

**Comment by stefanloerwald** [comment]
Hi @kaizen365,

Thanks for this contribution. Could you please add unit tests to this PR?

Thanks
Stefan

**Comment by kaizen365** [comment]
No problems @stefanloerwald will do (on my windows machine) - Looks like the tests don't run under apple silicon interestingly enough..

**Comment by kaizen365** [comment]
[x] Added Unit Tests
[x] Unit Tests pass under MacOS Apple Silicon

TODO:
[ ] Unit Tests pass under Windows (Not Tested yet)
[ ] Integration Testing - Confirm Tool working with IgnoreFiles and IgnoreFolders

**Comment by kaizen365** [comment]
- [x] Added Unit Tests
- [x] Unit Tests pass
  - MacOS Apple Silicon * **Unable to Run / Tests do not work under MacOS Apple Silicon**
  - [x] Linux x64
  - [x] Windows x64
- [x] Integration Testing - Confirmed Dot Net CLI Tool working with IgnoreFiles and IgnoreFolders
  - [x] MacOS Apple Silicon
  - [x] Linux x64
  - [x] Windows x64

**Comment by kaizen365** [comment]
@stefanloerwald for your review, thanks!

File: README.md [hunk 1]
```diff
@@ -43,7 +58,7 @@ The underlying SASS compiler is changed from libsass to dart-sass. This is a nec
       //"Precision": 5, // Remove this
       "RelativeUrls": true,
       "LineFeed": "Lf",
-      "SourceMap": false
+      "SourceMap": false,
```
Comment (stefanloerwald) [inline README.md:61]: Nit: Is this change intended? If so, why? If not, please undo.

File: README.md [hunk 2]
```diff
@@ -23,6 +23,21 @@ As a benefit, this implementation is cross-platform (x64 linux/win are tested, p
...
 
+#### Changes in version 3.2.Y
+
+Supports excluding certain files and folders when using recursive mode
+Simply add the following to CompilerSettings.json:
+
+```json
+  "CompilerSettings": {
+     "IgnoreFolders": ["./wwwroot/", "./bin/", "./obj/"],
+     "IgnoreFiles":
```
Comment (stefanloerwald) [inline README.md:34]: Consider making this a one-liner too, just like IgnoreFolders.

File: .vscode/tasks.json [hunk 3]
```diff
@@ -0,0 +1,54 @@
+{
```
Comment (stefanloerwald) [inline .vscode/tasks.json:1]: Please delete. .vscode should be in your .gitignore file ;-)

File: .vscode/launch.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{
```
Comment (stefanloerwald) [inline .vscode/launch.json:1]: Please delete. .vscode should be in your .gitignore file ;-)

File: WebCompiler/Compile/Compilers.cs [hunk 5]
```diff
@@ -29,7 +37,9 @@ public Compilers(Config config, string base_path)
         private static CompilationStep Compile(string file) => new CompilationStep(file);
         public CompilationStep TryCompile(string file)
         {
-            switch (Path.GetExtension(file).ToUpperInvariant())
+            if(SkipProcessingThisFile(file, _basePath)) return new CompilationStep(file);
```
Comment (stefanloerwald) [inline WebCompiler/Compile/Compilers.cs:40]: Please add curly braces.

File: WebCompiler/Compile/Compilers.cs [hunk 6]
```diff
@@ -15,9 +16,16 @@ public class Compilers
         private readonly Compiler place;
         private readonly Compiler cleanup;
         private readonly Compiler autoprefix;
+        private readonly List<string> _ignoreFiles;
```
Comment (stefanloerwald) [inline WebCompiler/Compile/Compilers.cs:19]: Nit: please keep the variable naming consistent with the rest of the file, i.e. without leading underscore.

File: WebCompiler/Configuration/Settings/SassSettings.cs [hunk 7]
```diff
@@ -1,4 +1,5 @@
-﻿using DartSassHost;
+﻿using System.Collections.Generic;
+using DartSassHost;
```
Comment (stefanloerwald) [inline WebCompiler/Configuration/Settings/SassSettings.cs:2]: Nit: order alphabetically

File: WebCompiler/package.csproj [hunk 8]
```diff
@@ -20,7 +20,7 @@
   <!--PackageId, Version-->
   <PropertyGroup>
     <PackageId>Excubo.WebCompiler</PackageId>
-    <VersionPrefix>3.1.3</VersionPrefix>
+    <VersionPrefix>3.2.2</VersionPrefix>
```
Comment (stefanloerwald) [inline WebCompiler/package.csproj:23]: Please undo this change. I'll release a new version right after merging with a new version number.

**Review by stefanloerwald — COMMENTED** [review]
Thanks @kaizen365 for this! Most comments are very nitpicky. I'd appreciate if you could fix these and then I'm sure we can merge early next week.

File: README.md (hunk 1 above)
Comment (kaizen365) [reply README.md:61]: good spot! might of been an merge/typo issue

File: .vscode/launch.json (hunk 4 above)
Comment (kaizen365) [reply .vscode/launch.json:1]: no worries - added to .gitignore.
(was using vscode - github codespaces :D)

File: WebCompiler/Configuration/Settings/SassSettings.cs (hunk 7 above)
Comment (kaizen365) [reply WebCompiler/Configuration/Settings/SassSettings.cs:2]: Good spot, I can remove this now as I moved List<string> IgnoreFiles/Folders up to CompilerSettingsCollection.

**Comment by kaizen365** [comment]
> Thanks @kaizen365 for this! Most comments are very nitpicky. I'd appreciate if you could fix these and then I'm sure we can merge early next week.

No problems at all @stefanloerwald! Updated accordingly

**Comment by stefanloerwald** [comment]
Thank you for the contribution! Deployment is on it's way.
